### PR TITLE
Add client deletion

### PR DESCRIPTION
**Pull request by hihuz**
In this PR we are adding the functionality to delete clients.

The flow is:
* on dashboard, click on a client to see details
* click on `delete client` it will open a confirmation modal
* if you click on the overlay or on cancel, nothing will happen
* if you click on `delete client` the client will be destroyed in db.

Here are some screenshots: 
Screenshot 2019-10-25 at 11 24 35

Screenshot 2019-10-25 at 11 25 22

### CODE REVIEW

File: console-src/src/enhancers/withOAuthClients.tsx [hunk 1]
```diff
@@ -90,8 +90,29 @@ class OAuthClientsProvider extends Component<Props, State> {
...
-    this.props.kontistClient.graphQL.rawQuery(deleteClientMutation, payload);
+  deleteClient = async (payload: DeleteOAuthClientPayload) => {
+    this.setState({
+      isLoading: true
+    });
+
+    const {
+      // Currently rawQuery return type does not include client mutation results
+      // @ts-ignore
+      deleteClient: client
```
Comment (johannespfeiffer) [inline console-src/src/enhancers/withOAuthClients.tsx]: Why do we have `client` as a return type here? Was expecting it to match `Promise<void> | void`?

Comment (hihuz) [reply console-src/src/enhancers/withOAuthClients.tsx]: Hmm you're right it's not particularly useful considering we already have the client id in the same context, I have removed it here: 0082e6162f110e64014741e36e015c3617bbd432

Ideally I would also remove it from the query entirely, but the type in our schema currently do not allow it and returns a non nullable client: 
Screenshot 2019-10-25 at 11 55 24

I will make a small PR against backend to fix that

Comment (hihuz) [reply console-src/src/enhancers/withOAuthClients.tsx]: Hmm apparently making it optional is not as simple as I thought, even if I go `@Mutation(returns => Client, { nullable: true })`, the mutation still expects me to specify some client fields as return value 🤔 

Would you be fine if we leave it like so for the time being and try to find an improvement for it later on?

Comment (johannespfeiffer) [reply console-src/src/enhancers/withOAuthClients.tsx]: Sure

**Pull request merged**